src/3d/symbols/qgspolygon3dsymbol_p.cpp

```diff
@@ -119,13 +119,6 @@ Qt3DRender::QGeometryRenderer *QgsPolygon3DSymbolEntityNode::renderer( const Qgs
     if ( QgsWkbTypes::isCurvedType( geom.geometry()->wkbType() ) )
       geom = QgsGeometry( geom.geometry()->segmentize() );
 
-    if ( !geom.isGeosValid() )
-    {
-      // invalid geometries break tessellation
-      qDebug() << "skipping invalid geometry" << f.id();
-      continue;
-    }
-
```

Comment:
Hmm - this is an important check that ensures that the data we pass to tessellation will not crash triangulator. From what I understand you removed it because the isValid() check ignores Z coordinate and thus vertical polygons (walls) are reported as invalid. But it is commonly the case that people use 2D-only polygons and some of them are not valid. We need to figure out how to keep both tribes happy...

Reply:
According to my tests with 3D data, vertical walls are not the only cases where GEOS was considering geometries as invalid. It, also, considers as invalid all multi-surfaces, for instance. This is, also, an issue for 3D solids used commonly for such 3D representations. Therefore, GEOS validation is highly unreliable when dealing with 3D data.

I believe we can still render GEOS invalid geometries on 3D, same as they are rendered in the original 2D canvas anyway. I think there are enough tests on the tesselator, now, that enables as many as possible geometries to be processed, without causing a crash.

Reply:
@wonde-sk, @liberostelios  validity an robustness of the algo. are two different things, validity being more strict (i.e. validity ensures no crash).

@liberostelios, I'm not sure how robust the sweep line algo. is with degenerate geometries, but bad things may happen (division by zero, or worse: infinite loop)

I don't think we must ensure complete SFS validity for polyhedral surface (because the topology check for POLYHEDRALSURFACE is a real pain and out of the scope of this PR), but ensuring that "rotated" polygons are geos-valid should be easy enough (although maybe time consuming).

Reply:
We could, possibly, validate with GEOS the individual polygons that are part of a multi-geometry, but as you said I think this is too time consuming. It would be useful to test the code with more data. I have tested it against a dataset which is popular for being pretty bad and with many issues and with a [huge](https://3d.bk.tudelft.nl/download/3dfier/Amsterdam.imgeo.zip) dataset of 5GBs which should expose many degenerate cases.

Reply:
if the algorithm behaves itself with degenerate cases, I do agree to skip validation.